src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWP.cs

```diff
@@ -61,24 +68,15 @@ public void InitializeAppInfo(string installedLocation)
 
             if (hResult == Hresult.Ok)
             {
-                var apps = new List<UWPApplication>();
-
-                List<IAppxManifestApplication> appsViaManifests = AppxPackageHelper.GetAppsFromManifest(stream);
-                foreach (var appInManifest in appsViaManifests)
-                {
-                    var app = new UWPApplication(appInManifest, this);
-                    apps.Add(app);
-                }
-
-                Apps = apps.Where(a =>
+                Apps = AppxPackageHelper.GetAppsFromManifest(stream).Select(appInManifest => new UWPApplication(appInManifest, this)).Where(a =>
                 {
                     var valid =
                     !string.IsNullOrEmpty(a.UserModelId) &&
                     !string.IsNullOrEmpty(a.DisplayName) &&
                     a.AppListEntry != "none";
 
                     return valid;
-                }).ToArray();
+                }).ToList();
```

Comment:
Can't Apps be just IEnumerable? If so, you don't need to call ToList here, it is generally slow.

Reply:
Since this is a public property on an object i think it would be good to enumerate it once?